### ISSUE TITLE
Custom section IDs get transformed to lower case

### ISSUE THREAD

**Issue by fritzmg** [contao/core-bundle#639]
Consider this scenario:

1. You have a running Contao `4.2.5` installation.
2. You are using custom sections in the page layout.
3. These custom sections are called `customSection1` and `customSection2` for example.
4. By default, Contao generates a markup for these custom sections, using their names as IDs: 
    ```html
    <div id="customSection1">
      …
    </div>
    ```
5. When updating to Contao `4.3.0` and running `/install.php`, these custom sections get transformed to the new system:
    ![cs1](https://cloud.githubusercontent.com/assets/4970961/20754671/4ff5b748-b70c-11e6-92b0-dffdaa3e13de.png)
6. However, when you now click _Save_ for the first time in the page layout, the IDs of the custom sections will be forced to lowercase:
    ![cs2](https://cloud.githubusercontent.com/assets/4970961/20754730/950ddcac-b70c-11e6-8014-f73cba3fcc15.png)
7. Which will generate the following HTML output now:
    ```html
    <div id="customsection1">
      …
    </div>
    ```

This might break the page layout, since the CSS would still reference the old, camel case IDs for example.

**Comment by ausi** [comment]
It’s not only a problem with CSS referencing the IDs, the assignments of modules and articles get lost too.

I think we should allow uppercase letters in the IDs to be backwards compatible.

**Comment by Seefahrer** [comment]
Does revocation of the alterations in a.m. files preliminary solve the problem?

**Comment by leofeyer** [comment]
Fixed in dcf46cabd0bdeca8d1aef7a1a46925160a707f4b.
